src/Umbraco.Core/Composing/ReferenceResolver.cs

```diff
@@ -65,7 +65,7 @@ public IEnumerable<Assembly> ResolveAssemblies()
                     if (assemblyName.FullName.StartsWith("Umbraco."))
                         continue;
 
-                    var assembly = Assembly.Load(assemblyName);
+                    var assembly = Assembly.LoadFrom(dll);
```

Comment:
The reasoning was because `Assembly.Load(assemblyName);` could not load the SqlCE assembly

Reply:
I think we need to revert this, loading LoadFrom vs Load are 2 different things into 2 different assembly load contexts. The Assembly.Load uses fusion binding and we definitely don't want to muck with this when loading plugins. Here's what I propose... this could be done after this PR is merged too, let me know what you think. I would also be more than happy to do this stuff (i.e. I'd like to)

* We revert all of the TypeFinder, etc... changes in this PR that deal with loading in SqlCe
* We manually load in SqlCe in a different way than is done here and we don't need to use TypeFinder for it. SqlCe might need to load with Assembly.LoadFrom because it's not actually compatible but we should not be loading in compatible assemblies that way

Reply:
Now SqlCE is manual loaded, and we just ignore if it is not avaiable. The implementation works if there is no project reference to the `Umbraco.Persistence.SqlCe` project. 

In this case, we also "remove" the fast install option, so you need to go to the next step, and type in db info. On the other hand, if SqlCE is available, the fast install is still possible.
SqlCe is available if all the required dlls are in the bin folder. (e.g. by making a reference to the  `Umbraco.Persistence.SqlCe`  project)